discovery_engine/lib/src/domain/feed_manager.dart

```diff
@@ -278,47 +297,81 @@ class FeedManager {
     final pref = SourcePreference(source, PreferenceMode.excluded);
     await _sourcePreferenceRepository.save(pref);
 
+    if (cfgFeatureStorage) {
+      final sources = await _sourceReactedRepo.fetchAll();
+      await _engine.addExcludedSource(sources, source);
+      return EngineEvent.addExcludedSourceRequestSucceeded(source);
+    }
+
     await _updateEngineSourcesOnAdd();
     return EngineEvent.addExcludedSourceRequestSucceeded(source);
   }
 
   /// Removes a source to excluded sources set.
```

Comment:
```suggestion
  /// Removes a source from the excluded sources set.
```
:art:  but we could prob make this and others here consistent with simply one of:
```
Adds (Removes) a(n) trusted (excluded) source.
```